frontend/src/metabase-types/api/card.ts

```diff
@@ -197,7 +192,7 @@ export type VisualizationSettings = {
 
   // Series
   "graph.dimensions"?: string[];
-  "graph.metrics"?: string[];
+  "graph.metrics"?: (string | null)[];
```

Comment:
In `getDefaultColumns` we have the possibility that `graph.metrics` can be `null` in the array. I'm not sure if this is totally correct but that's what I've gathered.

Reply:
hm.. @alxnddr is the git blamer 😄 could you verify this one if you have a chance? from what I understand from `getDefaultColumns` in `frontend/src/metabase/visualizations/shared/settings/cartesian-chart.ts`, the return type is:
```
{
  dimensions: (string | null)[];
  metrics: (string | null)[];
  bubble: string | null;
}
```

We use `getDefaultColumns` here in `getDefaultMetrics`:

```
export function getDefaultMetrics(
  rawSeries: RawSeries,
  settings: ComputedVisualizationSettings,
) {
  const [{ card }] = rawSeries;
  const prevMetrics = settings["graph.metrics"] ?? [];
  const defaultMetrics = getDefaultColumns(rawSeries).metrics;
  if (
    prevMetrics.length > 0 &&
    defaultMetrics.length > 0 &&
    defaultMetrics[0] == null
  ) {
    return prevMetrics;
  }
  return defaultMetrics.slice(0, getMaxMetricsSupported(card.display));
}
```


You can see when we're returning `prevMetrics` we're explicitly checking for a `null` at `defaultMetrics[0]` and return `prevMetrics` if there is a `null`. _But_ if one of the other conditions fails, we're not necessarily prevented from returning a `null` value here.

That's why I used the `null` value but I really would love to not have it too. Just need to make sure so we can make the appropriate typings.

Reply:
I was playing around with the `getDefaultMetrics` function on the core app (i.e. console logging and seeing what would pop up). It seems like `null` values really only occur when we're trying to use a visualization that doesn't necessarily fit the data. for example, trying to use the `bar` or `line` visualization on the full `Feedback` table - the X axis isn't defined so I get `[null]` as the value from `getDefaultMetrics`. The Y axis (`graph.dimensions`) is an empty array.